graphql_compiler/macros/__init__.py

```diff
@@ -104,7 +122,25 @@ def get_schema_with_macros(macro_registry):
     Returns:
         GraphQLSchema with additional fields where macroe edges can be used.
     """
-    raise NotImplementedError()  # TODO(bojanserafimov) implement
+    schema_ast = parse(print_schema(macro_registry.schema_without_macros))
+    for definition in schema_ast.definitions:
+        if isinstance(definition, ObjectTypeDefinition):
+            if definition.name.value in macro_registry.macro_edges:
+                macro_edges = macro_registry.macro_edges[definition.name.value]
+                for macro_edge_name, macro_edge_descriptor in six.iteritems(macro_edges):
+                    type_at_target = get_type_at_macro_edge_target(
+                        macro_registry.schema_without_macros,
+                        macro_edge_descriptor.expansion_ast)
+                    arguments = []
+                    definition.fields.append(FieldDefinition(
+                        Name(macro_edge_name), arguments,
+                        NamedType(Name(type_at_target.name)), directives=[]))
```

Comment:
I think for this PR this is fine, but before we merge this into `master` I think we should add a `@macro_edge` directive here, to signal that the edge is from a macro. This is important since macro edges have limitations on e.g. recursion and optionals, and the user should have a way to know that ahead of time, instead of only finding it out once they write a query and it fails to compile.

Reply:
Will do in the next PR